impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java

```diff
@@ -80,6 +81,7 @@ public class KafkaClusterConfig {
     private static final String SCRAM_256_SASL_MECHANISM_NAME = "SCRAM-SHA-256";
     private static final String SCRAM_512_SASL_MECHANISM_NAME = "SCRAM-SHA-512";
     private static final Pattern PRE_KAFKA_39 = Pattern.compile("^3\\.[0-8]\\..*$");
+    private static final Pattern POST_KAFKA_4_1 = Pattern.compile("^4\\.[1-9][0-9]*\\..*$");
```

Comment:
Post 4.1 to means means 4.2 onwards.  Do you mean post 4.0?

Reply:
Maybe inverting it to be like the existing 'pre-3.9' language would be less ambiguous.  So we would apply the old logic to pre-4.1